ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/adapter/ReferenceAdapter.kt

```diff
@@ -338,14 +340,22 @@ class ReferenceAdapter : PartnerAdapter {
         context: Context,
         request: PartnerAdLoadRequest,
     ): PartnerAd {
-        val ad =
-            ReferenceFullscreenAd(
-                context, request.partnerPlacement, if (request.format == AdFormat.REWARDED) {
-                    REWARDED
-                } else {
-                    INTERSTITIAL
+        val ad = ReferenceFullscreenAd(
+            context = context,
+            adUnitId = request.partnerPlacement,
+            adFormat = when (request.format) {
+                AdFormat.INTERSTITIAL -> INTERSTITIAL
+                AdFormat.REWARDED -> REWARDED
+                else -> {
+                    if (request.format.key == "rewarded_interstitial") {
+                        REWARDED_INTERSTITIAL
+                    } else {
+                        PartnerLogController.log(LOAD_FAILED)
+                        throw IllegalArgumentException("Unsupported ad format: ${request.format}")
```

Comment:
woah... let's just return Result.failure. the return type can be Result<PartnerAd> to facilitate this.  this seems excessive...